### DISCUSSION TITLE
meta-hours #2

### DISCUSSION THREAD

**Discussion by yangwao** [kodadot/nft-gallery#1699] [Meta-Hours]
This is briefly preparation for upcoming Meta-Hours, further edits might be required!

![i-see-dead-people-ghostbusters-meme-Edited](https://user-images.githubusercontent.com/5887929/148239937-a1214cf2-9184-40e8-839a-c11606bd6559.jpeg)

- despite i had good body battery, I feel less alert. I've encountered bit burnout that might be case I've probably i didn't been motivated reply to your dm soonest possible nor at all. But with more people on board, load on my brain will be more distributed on other people and that's best gift I can recieve to be honest.
- Since our last metahours i would say we've made fair amount progress despite holidays in the middle and i really appreciate those who invested their time and exchange it for new experience plus some small compensation on the top. [Namely @xylodrone made huge issue](https://github.com/kodadot/nft-gallery/issues/1619) where we will be heading to [KodaDot v3](https://github.com/kodadot/nft-gallery/issues/1619) with focus on few issues and she played well as temporary technical product manager, crafted new design and putt all issues we have in backlog at one place. Then there is Beta where we had tons of issues while being running into finish line to screw 🪛 few bits here and there as various design stuff was still not there. 

![image](https://user-images.githubusercontent.com/5887929/148245393-eb4fd2a2-3231-4d24-b53e-89dd2f51c0fd.png)

![image](https://user-images.githubusercontent.com/5887929/148245417-f4331d5f-e0b3-4aeb-864e-1d65d2cde625.png)

- i would thank to every developer who contributed this codebase, in average monthly cycle we had up to 15 ppl in average, totally 19 which is fair amount to notice and coordinate. Definitely sounds like big team of people, yet we are starting synchronous and asynchronous efforts to deliver stuff faster, without depending on timezone as our contributors are based all around world, that's really great. We set to have our async-firsr culture and no-meetings as they're not needed, except our internal bi-weekly deep calls with internal projects. In monthly averages we are hovering above 90 PR per month, to be exact hitting 100 PR per 30days which is impressive. Now checking stats we are at total 106 PRs which is even much better and hopefully we can be steady on this number, we don't have illusion chasing this number as it more shows our pace of change which is I would say brutal comparing to other repositories other there and so far I haven't seen this project who iterates this fast in the public and probably we will be bit faster when we publish few more issues through our investor network. Imagine. Now we are flying fast under radar, literally in stealth. Yeah we have over hundred of stars and forks, 120/120 iirc and we aren't stopping and forks are on steady rise. When you would want to compare us against any other web3 repository have in mind that their changes are mostly coming from Readme.md not from fundamental changes of applicaition itself and this is crucial part. Each developer and contributiors who touched our code had need to deeply understand how we work etc. To zoom out, we call this slow and hidden network effects where building moat of contributors would help as ever running engine of stuff. Like autonomous car which serves your citizens 24/7 and producing results without needing to recharge. We will still need work on way to find infinite ⛽ but we have already something, which could be potentially in form of $koda token where we will add extrinsic motivation for our contributors except nowadays $ksm which we will see total payouts made and you'll be blushed away how efficient it is comparing to results. Thus in result, future and early contributors could benefit from exposure of success of KodaDot they're making which I guess good. 

### Why I think developers are contributing to KodaDot?
- They can choose contribute more or switch and go elsewhere, we literally don't hold anyone and that's the spirit, nowadays contributors choosen us for few things i guess. 
-- **One** is size of the bounty because yes people need pay the bills
-- **Second** way how to earn something on the top, I'm aware that everyone has side hustle and gig economy is way how we operate, we don't hold anyone on full-time nor tracking hours. If you spend to make this issue with three hundreds bucks  in five minutes or five hours, it's solely on the contributor per se and how experienced they're are. 
-- **Third** is probably feeling of belonging as people meet their friends bc of work and shared goals to create very small religions and tribes and I guess we trying everything to have right culture in KodaDot. 
-- **Four** is probably that we've generated that amount of challenging issues that competitive coders are simply attracted to crack on something unusual than corporate job has can offer. 
-- **Fifth** is probably very important for one, it's public proof of contribution, because we are open source. When we've tried to hire people and been over 200+ applicants, they send me their resume, I mean what I can do that with piece of pdf? I was merely looking on their GitHub, we we're meeting people all around world and I have zero idea of their solid experience. Only truthful source for me was their public GitHub and gitlab, thus when you contribute to KodaDot, you can be definitely assure they will hire you elsewhere. And yes it happend already, got three DMs of people asking me about devs who contributed to Kodadot, who signed to their company. -- **Sixth** is probably something I would say are glitches, same as our code has challenging issues, we got it back by bug-hunting and looking for ghosts, yes we've got 👻 ghost busters issue recently so whenever you feel lucky, we have few ghosts waiting for you in our code. Like most funny is, how the heck we managed to have part of Readme of our repository in og: description. And others things like someone made accidentally wrong merging and removed same thing again over which was added in previous PRs and nothing wrong about it because at pace of adding new stuff we can easily loose something important is we've noticed it's gone or do

![Screenshot 2021-12-28 at 14 17 56](https://user-images.githubusercontent.com/5887929/148245977-9b945ff6-ff4b-4355-8f3a-d8424b52a350.png)

- namely I would thank to our super early contributors **@roiLeo** and his evercoming cosmic rays with docker which makes his environment works and sometimes doesn't - we have issue to make KodaDot working in docker so lets come over to help us if you're into devops. Then we have @prachi00  which is our so far only girl developer and I'm super excited to see how she is learning and accomplished harder and harder issues and I really appreciate her communication in our code reviews. Then we have **@rhaicode**  and **@jbienesdev** who helped us delivery few stuff like new charts engine, working on payout leaderboard and will see what kind of hacking they would to be part of. **AndriiHrichnuk** is amazing senior hacker as he participated through our Gitcoin issue to hack on previews and SVG for beloved NFTs and now he is hacking on our next stuff we call Portcullis, which would be NFT-gating as we want to avoid spammy people and raise bar quality though in our rooms and rooms of kusama ecosystem, we will be sharing updates on this, till then you can subscribe to private beta with your discord to help us test stuff to be perfect. To our recent grinders who are frequently appearing in hall of fame of PR contributors is @kkukelka. [He made himself famous with various visual updates](https://github.com/kodadot/nft-gallery/pulls?q=is%3Apr+author%3Akkukelka+is%3Aclosed), like namely authored history component, working PWA and grinding on new kind of stuff. I like his superior responsiveness to various issues!

![image](https://user-images.githubusercontent.com/5887929/148246318-a6e53073-d176-4754-9ef1-9625de15a4cc.png)

- speaking of contributors we have recently eagle 🦅 from Iceland (are there any eagles at all? I urban legend says they have no trees) and his name is damsky or cryptodamsky, he is helping us on community operations, lurking around in our Discord to help us drive stuff forward. Like I received testing KodaDot thirts high quality print and seems we will be having merchandise!
- Small ad, if you've been early or recent contributor to the KodaDot and you're eglible for tshirt shout out to our shirt drop channel on Discord for cotton award, we would say thank you in this form. Next is that we want to send tshirts to people who boosted our Discord so we can have more community lingo aka stickers, higher audio quality for calls etc, we plan to give thirts to first 30 people who boosted our server so please stop by with your Discord Nitro and participate to help us 

![image](https://user-images.githubusercontent.com/5887929/148245294-6e7180b7-9c21-46ba-a69a-1ed30f459b7d.png)

- [we've added recently to our Readme.md](https://github.com/kodadot/nft-gallery/blob/main-nuxt/README.md) to bit make better overview from high altitude what's we are up to. We have showed how looks in our repository to bit see for others how complex KodaDot is becoming, hope it would help new developers orient way better and experienced help to see what we could improve, reduce, recycle and reuse to make our app faster and more simplier in design. Then we have chart contributors over time where we can see steady rising line of new talent forking our repository for using on their own or contributing  back to upstream which is really much needed to help others kick start their idea much better. Then i've recently generated preview how you can participate to help us on continuous visualization of our repository to help other understand our growth and impress others through visual stimulus as not everyone understands what does it mean having pull request, added or removed lines, commits etc

![image](https://user-images.githubusercontent.com/5887929/148247035-4b2b4bfa-018f-482a-975f-7621bb11c394.png)

- we have [recently contest issue to visualize repository activity](https://github.com/kodadot/nft-gallery/issues/1661), like graph of issues, commits, by person etc, your creativity is limitations in this and best would be rewarded for visualization,we have award up to 1k$ to make visualization of our repository to show other people who don't understand numbers what amount of effort you're putting into this to better understand what amount of time, experience and brain power was invested in KodaDot

![image](https://user-images.githubusercontent.com/5887929/148246588-df45ff29-b71c-4889-93ee-76dd7a9bfd43.png)

- we are hiring https://kodadot.xyz/jobs all different kind of people. Right now we are focusing to get on board Developer Advocate aka developer relationship manager to throw hackathon 
- we are setting up hackathon with up to 50k in bounties plus expecting sponsorship by various other investors and other technological partners, you might notice hackathon label we have in our repository and feel free to add yours. We will have devpost where you could register, setup channel where you can look for your partner in crime for hacking stuff. Teams would have higger chance to win bounties as we aim to cooperate on the stuff. Hence if there will be solo hackers they can win something! 
- categories could be best team up to 2k, where solo hacker can get up to 500, we will have various tracks
- next meta-hours we would like to try on bi-weekly basis or whenever we have accomplished something we could share with our community
- probably that's it, I'll finish article on numbers which you could find at https://medium.com/kodadot thanks xylodrone for proof reading
- next,  [we will have article is kkukelka took role to write article about difference between our legacy app like version you see today vs beta we've been grinding recently.](https://github.com/kodadot/nft-gallery/issues/1654) You'll definitely notice speed improvements and meanwhile to not hold you off that long time, we decided postpone some issues to v2.2 which will be rolled out later like namely it's hard to say for me, but for sake of UX, we are decide to change architecture and cache your JPEGs on CDN, hence IPFS reference is still there and it would be pinned to IPFS and Arweave we need to improve end-user experience. This is btw what we originated idea with Permafrost and we successful presented this idea and we are looking for founder member there! It's definitely still impressive idea 💡 and we are experienced and knowledgeable to lead someone on this.
- Despite I haven't got greatest sleep (ss67), probably I'm bit over-trained from bike, I see I've generated fair amount of text while starring to the wall while having infrared companion starring at me.
- Regarding [Metaprime](https://twitter.com/metaprime_net), we are looking for spatial startups in any form and we would like to onboard them. We will have definitely grants there to help them drive their idea &c and nurture right network effects together as we are quite believers into [SpatialPunk](https://twitter.com/metaprime_net)
- speaking of culture 🧫 we're looking recently into biohacking and various other topics to bring and explore other use cases of NFTs together. 
- Despite I haven't got greatest sleep (ss67), probably I'm bit over-trained from bike, I see I've generated fair amount of text while starring to the wall while having infrared companion starring at me.
- this Metahours I really try not to push stuff nor anyone as I can get a lot of people are recovering from Holiday and covid, maybe both and I'm aware people might get chill so whoever wants to speak next time, show off their contribution, let me know
- to add to very last, you can find our recent progress always here for Beta stuff https://github.com/orgs/kodadot/projects/1/views/1 and that's why we are open about our development, this is our only roadmap to follow

![Screenshot 2022-01-05 at 11 30 25](https://user-images.githubusercontent.com/5887929/148245697-33464c87-8673-47d6-929b-ff129c5dd9aa.png)

- just on very last add, we've been today listed in the [awesome-nuxt](https://github.com/nuxt-community/awesome-nuxt/pull/497) repository and [awesome-vue](https://github.com/vuejs/awesome-vue/pull/3832) which I'm taking as stepping milestone and wider public recognition of important of KodaDot for the NFT ecosystem! Tell your friends we have bounties and rewards for them! Whenever you have suggestion where we should be listed as opensource projects based on libraries we are using, suggest us to the [issue 1208](https://github.com/kodadot/nft-gallery/issues/1208)

### Next in agenda

- @vikiival will tell about 
   - statemine 
   - beta
   - new indexer (SubSquid)

- @XyloDrone would talk about [further upgrades](https://github.com/kodadot/nft-gallery/issues/1619) and things

- @cryptodamsky will talk about onboarding experience and contest

### KodaDot contributors 

![](https://media.giphy.com/media/3o7WIP2bOnnVXs4xz2/giphy-downsized.gif)

![](https://media.giphy.com/media/5rlkqJ8qbSiqI/giphy.gif)

**Comment by vikiival** [comment]
MetaHours @vikiival

- re: statemine
    - to successfully pass the coucil proposal we have to finishi article
    - In general the article is the review of the NFT pallet
    - [beta.kodadot.xyz/statemine](beta.kodadot.xyz/statemine) 
- Indexers
    - We sucessfully implemented subsquid as another indexer
    - in comparison super fast indexing
    - and also we are able to define our own queries
    - already in prod on beta for the /spotlight and /series-insight
    - of course the code is open source and you can find it under kodadot organization on github
    - updates are super fast so if you would like to see some new views in koda open issue 

- As we speak about Beta
    - Porting last components from the SPA - yesterday we closed about 7 issues only regarding to that
    - still some ghost around: forming ghostbusters squad 
    - Happy if you would test it ❤️ we have a bug bounty program
    - already with v3

**Comment by XyloDrone** [comment]
### UX/UI Agenda: 

**Obstacles** 

- A lot of issues in Github | difficult to manage and navigate

- design elements: color contrasts, color pallets component sizes

- onboarding procedure 

- mobile version 

- making the platform your own through a personalised feature that allows tracking the activity and prices of artworks, collectors and artists 


- naming  and positioning of features and pages 


**Solutions** 

1. Composing all UX/UI GitHub issues within 1 data architecture 

2. Updating colors and sizes for a more clean look 

3. Gameify the onboarding procedure, which would start on the landing page 

4. Getting started on separate designs for mobile, as NFT experiences are very different on desktop and handheld devices. If we think of our digital assets as showcases within a gallery, they will look very differently depending on the device displaying these + their use case 

5. Prototyping a customised collector/artist dashboard which showcases activity of the works, creators and collectors we like or are in touch with. This will result in much better management tooling which could potentially result in further users 

6. Naming certain features that sound confusing to newcomers, primarily within the “extra section”(eg admin panel, which is really a mass mint or mass air drop tool that is more fitting within the pages of minting and profile dashboards)

**Comment by damskyftw** [comment]
Hi, ye i kinda wanna say few words about my presence on kodadot. It's about one month since i started in this community and it's been hell of a ride ever since. We've got ton of issues which even non technical people can do and that will be my primary focus for the time being. I can see a big gap between people who would want to contribute and the actual process of contributing. Main reasons why i think that is is that crypto is still relatively young and the technology itself is still not yet that normie-friendly  but that's why there are people like me who can make it more accessible to everyone by making tutorials, or helping them through.

Main stuff to be done is definitely the onboarding experience for non tech savy users and since we have a beta now which is quite substantial i would like anyone to work on it, since there is plenty of bounties and simple quests.

Also as some of you may have seen, i'm in charge of sending t-shirts, so anyone who's eligible for t-drop either by boosting the server or being a long term contributor, just hit me up.

Next thing i wanted to talk about is gif meme contest. We wanna push more kodadot gifs to giphy and tenor and hence we decided to make a contest. The contest will be rolling, means that it's not gonna have a deadline and we are simply going to track traction of each gif with responses on them. Keep in mind that it's still experimental as KodaDot landscape and therefore the rewarding mechanism still stands to few persons, but we are grateful to all contributions and looking forward to see some meme gifs from you guys! There's already one pushed on giphy i think so you can have a look how that looks and do it in similiar fashion. Some guidelines to make sure everyone is satisfied
- Meme-gif should be uploaded to giphy/tenor
- Use KodaDot v3 presskit materials in the gif
- SFW content
- and last, just please no racist stuff or anything of that kind, we wanna keep it simple and friendly
- Happy to pay 0.5-1 ksm for good gif
- I heard first 3 contributors are eligible for t-shirt

Specific channel with annoncment and details will be created, so you can post your contributions there. :) Stay tuned.

![nft-minting-kusama-kodadot-ksm](https://user-images.githubusercontent.com/92786519/148253234-12ed2be9-0388-4a54-b330-41b3ef270d8d.gif)


You might've also heard me talking on the last episode of meta-hours about discord sticker contest, which i assume is going to be a little more time consuming, so we wanna basicly test out this voting-traction mechanism before we dive into something bigger.

**Comment by yangwao** [comment]
As been bit down and exhausted, I forget to mention that @rchovatiya88 and @martinloesethjensen are helping us as well! 
 
@rchovatiya88 is dedicated to XR stuff and @martinloesethjensen is grinding on background koda mobile. Hopefully, we will have something to share, especially on the eXtended Reality (kodaverse) track and mobile track, I'm looking at updates there in two weeks at [Meta_Hours_3](https://github.com/kodadot/nft-gallery/discussions/1710) 😇
